rtt/os/tlsf/target.h

```diff
@@ -41,10 +41,22 @@
 
 #include "../fosi.h"
 
-#define TLSF_MLOCK_T            rt_mutex_t
-#define TLSF_CREATE_LOCK(l)     rtos_mutex_init (l)
-#define TLSF_DESTROY_LOCK(l)    rtos_mutex_destroy(l)
-#define TLSF_ACQUIRE_LOCK(l)    rtos_mutex_lock(l)
-#define TLSF_RELEASE_LOCK(l)    rtos_mutex_unlock(l)
+#if defined(OROPKG_OS_MACOSX)
+// Use pthread directly for TLSF.
+// The macosx implementation of rt_mutex_t has some overhead to support
+// timed locking.
```

Comment:
Seemingly the timed locking is not required for `TLSF`, so is there a reason not to always use `pthread` everywhere for consistency? :thinking: